tests/test_ir.py

```diff
@@ -170,16 +168,13 @@ def test_op_clone_with_regions():
 @irdl_op_definition
 class SuccessorOp(IRDLOperation):
     """
-    Utility operation that requires a successor and has the IsTerminator
-    trait.
+    Utility operation that requires a successor.
     """
 
     name = "test.successor_op"
 
     successor: Successor = successor_def()
 
-    traits = frozenset([IsTerminator()])
```

Comment:
This looks suspicious to me.
If an operation has successor, it needs the terminator trait.

Reply:
test_block_not_branching_to_another_region should fail because of that (also because the first block is empty)